### PR TITLE
konvoy: Update docs for konvoy release v1.5.x-dk

### PR DESCRIPTION
**Pull request by dkoshkin**
## Description of changes being made

@russurquhart1 I think you have reviewed most of this as they were being merged into Konvoy, but Im going to highlight the changes:

- [ ] `install/bastion-hosts/index.md`
- [ ] `install-aws/advanced-provisioning/index.md`
- [ ] `install/install-gcp/*`
- [ ] `install/install-secured/index.md` - new page to describe an install on secure infra.
- [x] `install/ssh-configuration/index.md` you added this
- [ ] `reference/cluster-configuration/*` all these are auto generated and migrated from what was there previously
- [x] `reference/command-line-interface/*` these are autogenerated, you suggested the word changes
- [ ] `security/external-idps/api-reference/index.md` - these are mainly coming from upstream
- [ ] `security/letsencrypt/index.md`
- [x] `tutorials/splunk-integration/index.md` - you added this
- [x] `upgrade/upgrade-kubernetes-addons/index.md` - you suggested these changes.

### CODE REVIEW

File: pages/ksphere/konvoy/latest/install/bastion-hosts/index.md [hunk 1]
```diff
@@ -166,6 +166,43 @@ bastion:
     ansible_port: 22
 ```
 
+**Please Note:** If you are using the Bastion host feature with an internal facing loadbalancer for Kubernetes API [(`aws.elb.internal: true`)][aws_elb] in the `ClusterConfiguration` section of `cluster.yaml` and deploying from a machine outside the network of the cluster, there is a known issue where the deployment of the Addons will fail to connect to the Kubernetes API with the following example error:
```
Comment (russurquhart1) [inline pages/ksphere/konvoy/latest/install/bastion-hosts/index.md:169]:
```suggestion
<p class="message--note"><strong>NOTE: </strong>

If you are using the Bastion host feature with an internal facing loadbalancer, for Kubernetes API, [(`aws.elb.internal: true`)][aws_elb] in the `ClusterConfiguration` section of `cluster.yaml` and deploying from a machine outside the network of the cluster, there is a known issue. Deployment of the Addons fails to connect to the Kubernetes API with the following example error:</p>
```

File: pages/ksphere/konvoy/latest/install/bastion-hosts/index.md [hunk 2]
```diff
@@ -166,6 +166,43 @@ bastion:
...
+
+```bash
+STAGE [Deploying Enabled Addons]
+Get https://internal-konvoy-75d3-lb-control-123456789.us-east-1.elb.amazonaws.com:6443/api?timeout=32s: dial tcp 10.0.67.165:6443: i/o timeout[ERROR]
+
+Error: failed to deploy the cluster: Get https://internal-konvoy-75d3-lb-control-123456789.us-east-1.elb.amazonaws.com:6443/api?timeout=32s: dial tcp 10.0.67.165:6443: i/o timeout
+exit status 1
+```
+
+Please use the following workaround while the issue is being addressed:
```
Comment (russurquhart1) [inline pages/ksphere/konvoy/latest/install/bastion-hosts/index.md:179]:
```suggestion
Use the following workaround while the issue is being addressed:
```

File: pages/ksphere/konvoy/latest/install/install-gcp/advanced-provisioning/index.md [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+For each [node pool][node_pools], users can customize the instance type for instances, in that node pool, using the `type` field.
+
+All available virtual machine types can be found [here][virtual_machine_types].
+
+## Instance volumes
+
+For each [node pool][node_pools], users can customize the instance volumes attached to the instances in that node pool.
+There are two types of instance volumes:
+
+* Root volume: this is the root disk for providing [ephemeral storage][ephemeral_storage] for the Kubernetes node (except container images if imagefs volume is enabled).
```
Comment (russurquhart1) [inline pages/ksphere/konvoy/latest/install/install-gcp/advanced-provisioning/index.md:86]:
```suggestion
* Root volume: This is the root disk for providing [ephemeral storage][ephemeral_storage] for the Kubernetes node (except container images if imagefs volume is enabled).
```

File: pages/ksphere/konvoy/latest/install/install-gcp/advanced-provisioning/index.md [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+* Imagefs volume: this is the dedicated disk for providing storage for container image layers.
```
Comment (russurquhart1) [inline pages/ksphere/konvoy/latest/install/install-gcp/advanced-provisioning/index.md:87]:
```suggestion
* Imagefs volume: This is the dedicated disk for providing storage for container image layers.
```

File: pages/ksphere/konvoy/latest/install/install-gcp/advanced-provisioning/index.md [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+In GCP, you can define already available [public images][public_images] as `imageID` or your own images from an available image self link.
+
+The default GCP Image used for Konvoy cluster deployments is the latest available `centos-7`
+
+Konvoy is tested with the `CentOS Linux 7` image.
+
+# Adding custom Terraform resources for provisioning
+
+You can provide custom `*.tf` resource files when provisioning.
+If you create additional resource files, they are used along with the default `*.tf` resource files during the provisioning process.
```
Comment (russurquhart1) [inline pages/ksphere/konvoy/latest/install/install-gcp/advanced-provisioning/index.md:105]:
```suggestion
If you create additional resource files, they are used with the default `*.tf` resource files during the provisioning process.
```

File: pages/ksphere/konvoy/latest/install/install-gcp/index.md [hunk 6]
```diff
@@ -0,0 +1,377 @@
...
+navigationTitle: Install on Google Cloud Platform (GCP)
+title: Install on Google Cloud Platform (GCP)
+menuWeight: 20
+excerpt: Prepare for and install Konvoy on Google Cloud Platform (GCP)
+enterprise: false
+---
+
+<!-- markdownlint-disable MD004 MD007 MD025 MD030 -->
+
+This section describes the steps to prepare your environment and install Konvoy on Google Cloud Platform (GCP).
```
Comment (russurquhart1) [inline pages/ksphere/konvoy/latest/install/install-gcp/index.md:12]:
```suggestion
This section describes how to prepare your environment and install Konvoy on Google Cloud Platform (GCP).
```

File: pages/ksphere/konvoy/latest/install/install-gcp/index.md [hunk 7]
```diff
@@ -0,0 +1,377 @@
...
+
+## Prerequisites
+
+* The [gcloud][install_gcloud] command line utility
+* [Docker Desktop][install_docker] _version 18.09.2 or newer_
+* [kubectl][install_kubectl] _v1.15.4 or newer_ (for interacting with the running cluster)
+* A valid GCP account with [credentials configured][gcloud_sdk].
+  Make sure to set your `project` and `region` you want to use and export it
```
Comment (russurquhart1) [inline pages/ksphere/konvoy/latest/install/install-gcp/index.md:20]:
```suggestion
  Make sure to set your `project` and `region` you want to use, and export it
```

File: pages/ksphere/konvoy/latest/install/install-secured/index.md [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+navigationTitle: Install on secured machines
+title: Install on secured machines
+menuWeight: 37
+excerpt: Install on secured machines
+enterprise: false
+---
+
+<!-- markdownlint-disable MD004 MD007 MD025 MD030 -->
+
+The topics in this section guide you through the basic steps to prepare your environment and install Konvoy in an air-gapped environment.
```
Comment (russurquhart1) [inline pages/ksphere/konvoy/latest/install/install-secured/index.md:12]:
```suggestion
This section describes how to prepare your environment and install Konvoy in an air-gapped environment.
```

File: pages/ksphere/konvoy/latest/install/install-secured/index.md [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+**IMPORTANT** air-gapped installation is still in Beta and the process may change in the future.
```
Comment (russurquhart1) [inline pages/ksphere/konvoy/latest/install/install-secured/index.md:13]:
```suggestion
<p class="message--important"><strong>IMPORTANT: </strong>Air-gapped installation is still in Beta and the process may change in the future.</p>

```

File: pages/ksphere/konvoy/latest/install/install-secured/index.md [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+
+# Before you begin
+
+Before installing, verify that your environment meets the following basic requirements:
```
Comment (russurquhart1) [inline pages/ksphere/konvoy/latest/install/install-secured/index.md:17]:
```suggestion
Before installing, ensure that your environment has the following basic requirements:
```

File: pages/ksphere/konvoy/latest/install/install-secured/index.md [hunk 11]
```diff
@@ -0,0 +1,180 @@
...
+* [Docker Desktop][install_docker] version 18.09.2 or later
+
+  You must have Docker Desktop installed on the host where the Konvoy command line interface (CLI) will run.
+  For example, if you are installing Konvoy on your laptop, be sure the laptop has a supported version of Docker Desktop.
+
+* [kubectl][install_kubectl] v1.16.8 or later
+
+  To enable interaction with the running cluster, you must have `kubectl` installed on the host where the Konvoy command line interface (CLI) will run.
+
+* The `konvoy_air_gapped.tar.bz2` that will contain the required artifacts to perform an air-gapped installation.
```
Comment (russurquhart1) [inline pages/ksphere/konvoy/latest/install/install-secured/index.md:28]:
```suggestion
* The `konvoy_air_gapped.tar.bz2` that contains the required artifacts to perform an air-gapped installation.
```

File: pages/ksphere/konvoy/latest/install/install-secured/index.md [hunk 12]
```diff
@@ -0,0 +1,180 @@
...
+* Each control plane node should have at least:
+  * 4 cores
+  * 16 GiB memory
+  * 80 GiB of free space in the root partition, and the root partition must be less than 85% full.
+
+## Worker nodes
+
+* You should have at least four worker nodes.
+
+  The specific number of worker nodes required for your environment can vary depending on the cluster workload and size of the nodes.
```
Comment (russurquhart1) [inline pages/ksphere/konvoy/latest/install/install-secured/index.md:43]:
```suggestion
  The specific number of worker nodes required for your environment varies depending on the cluster workload and size of the nodes.
```

File: pages/ksphere/konvoy/latest/install/install-secured/index.md [hunk 13]
```diff
@@ -0,0 +1,180 @@
...
+
+* Firewalld is disabled.
+* Containerd is uninstalled.
+* Docker-ce is uninstalled.
+* Swap is disabled.
+
+## Installation
+
+On highly secured clusters you may need to modify the `cluster.yaml` file with additional options.
+See below for possible changes that may be applied in your cluster.
```
Comment (russurquhart1) [inline pages/ksphere/konvoy/latest/install/install-secured/index.md:65]:
```suggestion
See the sample file below for possible changes that may be applied in your cluster.
```

**Comment by dkoshkin** [comment]
The changes proposed here were added in https://github.com/mesosphere/konvoy/pull/1412/commits/25ba67554efd0b6207e4befae68913dc30e8aa9c

File: pages/ksphere/konvoy/latest/release-notes/index.md [hunk 14]
```diff
@@ -15,8 +16,134 @@ enterprise: false
...
+
+- Kubernetes `v1.16.8`
+- Calico `v3.13.1`
+- Go `1.13.8`
+- Ansible `2.7.16`
+
+#### Addon versions
+
+-   awsebscsiprovisioner: 0.5.0-1
+    - awsebscsiprovisioner: 0.5.0
```
Comment (samvantran) [inline pages/ksphere/konvoy/latest/release-notes/index.md:63]: so many of these are single item addons that have the base-tech version shown in the revision. Doesn't have to be this release but maybe for future we want to show only the revision (eg `0.5.0-1`) if the addon is only made of one chart/app?

**Pull request merged**